docs/refguide/api.md

```diff
@@ -172,36 +204,38 @@ Usage:
 
 
 ### `autorun`
-Usage: `autorun(debugname?, () => { sideEffect })`. Autorun runs the provided `sideEffect` and tracks which observable state is accessed while running the side effect.
+Usage: `autorun(() => { sideEffect }, options)`. Autorun runs the provided `sideEffect` and tracks which observable state is accessed while running the side effect.
 Whenever one of the used observables is changed in the future, the same sideEffect will be run again.
 Returns a disposer function to cancel the side effect. [&laquo;details&raquo;](autorun.md)
 
+**options**
+- **`debugname?: string`**: A name for easier identification and debugging
+- **`delay?: number`**: the sideEffect will be delayed and debounced with the given `delay`. Defaults to `0`.
+
 ### `when`
-Usage: `when(debugname?, () => condition, () => { sideEffect })`.
+Usage: `when(() => condition, () => { sideEffect }, options)`.
```

Comment:
effect is now optional, if no effect is specified, returns a (cancellable) promise

options:
- `name`
- `timeout`
- `onError`